### DISCUSSION TITLE
Updating Kubernetes Clusters installed with Helm

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5403] [FAQs]
I created a Kubernetes cluster using Helm as instructed here https://github.com/ant-media/helm some months back and therefore it is running with Ant Media Server Enterprise edition v2.5.3

As I see AMS v2.6.2 has been released and I would like to update my K8 cluster from AMS v2.5.3 to AMS v2.6.2.

How can I do this easily without losing my current configurations and without having to reinstall the K8 cluster?

Thank you.

**Comment by muratugureminoglu** [comment]
You can update Ant Media Server to the latest version by changing the image version in the Deployment files. Since all settings are stored in the MongoDB Pod, your configurations will be preserved as long as you do not delete the Pod.

You can do this in two different methods.

**Method 1:**

Run the following command to update the deployment files for both origin and edge:

```bash
kubectl edit deployment ant-media-server-origin -n antmedia
```

Find the following line, update it with the latest version number, and save the changes:

```yaml
image: antmedia/enterprise:2.6.2
```

**Method 2:**

Alternatively, you can use the following command as well:

```bash
kubectl patch deployment ant-media-server-origin -p '{"spec":{"template":{"spec":{"containers":[{"name":"temp-name","image":"antmedia/enterprise:latest"}]}}}}'
```

For a specific version, you can use the following command (replace the version number, for example, 2.6.2):

```bash
kubectl patch deployment ant-media-server-origin -p '{"spec":{"template":{"spec":{"containers":[{"name":"temp-name","image":"antmedia/enterprise:2.6.2"}]}}}}'
```

**If the new Pods are still in the "Pending" state after the above changes, you can reset the replicas using the following commands:**

```bash
kubectl scale deployment ant-media-server-origin --replicas=0 -n antmedia
kubectl scale deployment ant-media-server-origin --replicas=1 -n antmedia
```

This will recreate the Pods, and the new Pods will use the updated image.
